core/domain/summary_services.py

```diff
@@ -163,6 +163,52 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_exploration_dicts(
+        exploration_ids, editor_user_id=None):
+    """Given a list of exploration ids, optionally filters the list for
+    explorations that are currently non-private and not deleted, and returns a
+    list of dicts of the corresponding exploration summaries for collection
+    node search.
+    """
+    exploration_summaries = (
+        exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+
+    filtered_exploration_summaries = []
+    for exploration_summary in exploration_summaries:
+        if exploration_summary is None:
+            continue
+        if exploration_summary.status == (
+                rights_manager.ACTIVITY_STATUS_PRIVATE):
+            if editor_user_id is None:
+                continue
+            if not rights_manager.Actor(editor_user_id).can_edit(
+                    feconf.ACTIVITY_TYPE_EXPLORATION,
+                    exploration_summary.id):
+                continue
+
+        filtered_exploration_summaries.append(exploration_summary)
+
+    exploration_ids = [
+        exploration_summary.id
+        for exploration_summary in filtered_exploration_summaries]
+
+    searchable_exp_summaries = []
```

Comment:
Oh, yes. I had just been (erroneously, I think) following what you used below -- which seems to be exploration_summaries instead of filtered_exploration_summaries.